frontendUiTable/src/main/java/com/virtuslab/gitmachete/frontend/ui/table/GitMacheteGraphTableManager.java

```diff
@@ -178,14 +183,28 @@ public void run(ProgressIndicator indicator) {
    * Updates repository which is the base of graph table model. The change will be seen after
    * {@link GitMacheteGraphTableManager#refreshGraphTable()}.
    */
+  @SuppressWarnings({"IllegalCatch"})
   private void updateRepository(Path mainDirectoryPath, Path gitDirectoryPath, boolean isMacheteFilePresent) {
     if (isMacheteFilePresent) {
-      Try.of(() -> {
+      // This try-catch is a workaround caused strange behavior of Try.onFailure() that seems to rethrows exception
+      // (or something else happens)
+      // For state that caused unexpected behavior see tag `strange-vavr-try-behavior`
+      try {
         IBranchLayout branchLayout = updateBranchLayoutAndMacheteFilePath();
-        return gitMacheteRepositoryFactory.create(mainDirectoryPath, gitDirectoryPath, branchLayout);
-      })
-          .onFailure(e -> LOG.error("Unable to create Git Machete repository", e))
-          .onSuccess(repository -> repositoryRef.set(repository));
+        repositoryRef.set(gitMacheteRepositoryFactory.create(mainDirectoryPath, gitDirectoryPath, branchLayout));
+      } catch (Exception e) {
+        LOG.error("Unable to create Git Machete repository", e);
+        String exceptionMessage = e.getMessage();
+        VcsNotifier.getInstance(project).notifyError("Repository instantiation failed",
+            exceptionMessage != null ? exceptionMessage : "");
+        GuiUtils.invokeLaterIfNeeded(
+            () -> Messages.showErrorDialog(
+                exceptionMessage != null
+                    ? exceptionMessage
+                    : "Repository instantiation failed. For more information, pleas look at the IntelliJ logs",
```

Comment:
`please`

Reply:
Fun fact, `pleas` is plural of:

![image](https://user-images.githubusercontent.com/3383210/80514361-aed16880-8980-11ea-8915-4512eb9d85ef.png)

Reply:
Likely a word related to _please_ btw